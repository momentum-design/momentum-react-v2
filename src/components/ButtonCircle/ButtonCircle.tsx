import React, { Children, forwardRef, RefObject } from 'react';
import classnames from 'classnames';
import ButtonSimple from '../ButtonSimple';

import { DEFAULTS, STYLE } from './ButtonCircle.constants';
import { Props } from './ButtonCircle.types';
import './ButtonCircle.style.scss';

const ButtonCircle = forwardRef((props: Props, providedRef: RefObject<HTMLButtonElement>) => {
  const {
    children,
    className,
    color,
    disabled,
    shallowDisabled,
    ghost,
    outline,
    inverted,
    size,
    onlyTriggersTooltip,
    ...otherProps
  } = props;

  if (ghost && inverted) {
    console.warn('MRV2: Momentum does not support a ghost inverted ButtonCircle.');
  }

  if (onlyTriggersTooltip && (outline || inverted || color || disabled || shallowDisabled)) {
    console.warn(
      'MRV2: Momentum does not support a button that onlyTriggersTooltip and is outline, inverted, has color, or is disabled'
    );
  }

  const multipleChildren = Children.count(children) > 1;

  return (
    <ButtonSimple
      className={classnames(STYLE.wrapper, className)}
      ref={providedRef}
      data-color={color || DEFAULTS.COLOR}
      data-ghost={ghost || onlyTriggersTooltip || DEFAULTS.GHOST}
      data-multiple-children={multipleChildren}
      data-outline={outline || DEFAULTS.OUTLINE}
      data-size={size || DEFAULTS.SIZE}
      data-inverted={inverted || DEFAULTS.INVERTED}
      data-disabled={disabled || DEFAULTS.DISABLED}
      data-shallow-disabled={shallowDisabled || DEFAULTS.SHALLOW_DISABLED}
      data-only-triggers-tooltip={onlyTriggersTooltip || DEFAULTS.ONLY_TRIGGERS_TOOLTIP}
      aria-disabled={shallowDisabled || DEFAULTS.SHALLOW_DISABLED}
      isDisabled={disabled}
      {...otherProps}
    >
      {children}
    </ButtonSimple>
  );
});

ButtonCircle.displayName = 'ButtonCircle';

export default ButtonCircle;
