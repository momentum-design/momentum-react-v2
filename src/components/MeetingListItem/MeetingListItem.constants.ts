import { MeetingMarker } from './MeetingListItem.types';

const CLASS_PREFIX = 'md-meeting-list-item';

const DEFAULTS = {
  color: MeetingMarker.Transparent,
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
