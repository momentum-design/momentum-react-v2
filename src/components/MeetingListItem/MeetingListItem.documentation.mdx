`<MeetingListItem />` component provides a configurable container for a meeting (including active,
scheduled, recorded) list item.

# Structure

The structure of the component is as follows:

```
+--------+---------------+--------------+--------------+
| Border | Start Content | Main Content | Button Group |
+--------+---------------+--------------+--------------+
```

- Border: based on `color` provided
- Start Content: based on `image` provided
- Main Content: based on `children` provided
- Button Group: based on `buttonGroup` provided

## Simple example

```js
<MeetingListItem>
  <div>Date</div>
</MeetingListItem>
```

## Complex example

```js
<MeetingListItem
  buttonGroup={
    <ButtonGroup spaced>
      <ButtonHyperlink>Link</ButtonHyperlink>
      <div style={{ display: 'flex', alignItems: 'center' }}>
        17 <Icon autoScale name="participant-list" />
      </div>
      <ButtonPill color="join">Join</ButtonPill>
    </ButtonGroup>
  }
  color={MeetingMarker.AcceptedActive}
  image={<Avatar initials="TU" size={32} />}
>
  <div key="child1">Date</div>
  <div key="child2">Normal</div>
</MeetingListItem>
```
