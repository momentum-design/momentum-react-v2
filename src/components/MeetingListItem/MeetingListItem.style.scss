.md-meeting-list-item-wrapper {
  position: relative;
  margin: 0.25rem 0;
  overflow: hidden;

  &:not([data-color='Transparent']) {
    background-color: var(--mds-color-theme-background-secondary-normal);

    &:hover,
    &.hover {
      background-color: var(--mds-color-theme-background-secondary-hover);
    }

    &:active,
    &.active {
      background-color: var(--mds-color-theme-background-secondary-active);
    }

    &[data-disabled='true'],
    &.disable {
      background-color: var(--mds-color-theme-background-secondary-normal);
    }
  }

  &[data-color='GrayStatic'] {
    background-color: var(--mds-color-theme-background-secondary-normal);

    &.hover,
    &:hover,
    &.active,
    &:active {
      background-color: var(--mds-color-theme-background-secondary-normal);
    }
  }

  &[data-disabled='false'] {
    .md-meeting-row-content-middle-section {
      *:first-of-type {
        color: var(--mds-color-theme-text-primary-normal);
      }

      *:last-of-type:not(:first-of-type) {
        color: var(--mds-color-theme-text-secondary-normal);
      }
    }
  }
}
