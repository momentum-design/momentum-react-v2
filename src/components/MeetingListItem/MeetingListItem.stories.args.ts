import { commonAriaPressProps, commonStyles } from '../../storybook/helper.stories.argtypes';
import { MeetingMarker } from './MeetingListItem.types';

export default {
  ...commonStyles,
  ...commonAriaPressProps,
  children: {
    description: 'Provides the child nodes for this element.', // Description of this prop.
    // Control
    table: {
      type: {
        summary: 'ReactNode', // Explicit type of this prop.
      },
      defaultValue: {
        summary: 'undefined', // Default value when rendering this component.
      },
    },
  },
  isDisabled: {
    defaultValue: false,
    description: 'Determines if this item is disabled',
    control: { type: 'boolean' },
    table: {
      type: {
        summary: 'boolean',
      },
      defaultValue: {
        summary: false,
      },
    },
  },
  buttonGroup: {
    defaultValue: undefined,
    description: 'Provides the nodes for the end of the item',

    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: {
        summary: undefined,
      },
    },
  },
  color: {
    defaultValue: undefined,
    description: 'Provides the color status of this item',
    control: { type: 'select' },
    options: [...Object.keys(MeetingMarker)],
    table: {
      type: {
        summary: 'MeetingMarker',
      },
      defaultValue: {
        summary: undefined,
      },
    },
  },
  image: {
    defaultValue: undefined,
    description: 'Provides the nodes for the start of the item',

    table: {
      type: {
        summary: 'ReactNode',
      },
      defaultValue: {
        summary: undefined,
      },
    },
  },
};
