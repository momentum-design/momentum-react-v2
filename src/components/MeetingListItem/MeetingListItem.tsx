import React, { FC, forwardRef, useRef, RefObject } from 'react';
import classnames from 'classnames';

import { STYLE, DEFAULTS } from './MeetingListItem.constants';
import { Props } from './MeetingListItem.types';
import './MeetingListItem.style.scss';
import ListItemBase from '../ListItemBase';
import MeetingRowContent from '../MeetingRowContent';

const MeetingListItem: FC<Props> = forwardRef(
  (props: Props, providedRef: RefObject<HTMLLIElement>) => {
    const {
      children,
      className,
      color = DEFAULTS.color,
      isDisabled,
      buttonGroup,
      image,
      large,
      itemIndex,
      ...rest
    } = props;

    const internalRef = useRef();
    const ref = providedRef || internalRef;

    return (
      <ListItemBase
        isPadded
        ref={ref}
        className={`${classnames(className, STYLE.wrapper)}`}
        data-color={color}
        isDisabled={isDisabled}
        itemIndex={itemIndex}
        size={large ? 70 : 50}
        {...rest}
      >
        <MeetingRowContent
          color={color}
          isDisabled={isDisabled}
          buttonGroup={buttonGroup}
          image={image}
        >
          {children}
        </MeetingRowContent>
      </ListItemBase>
    );
  }
);

MeetingListItem.displayName = 'MeetingListItem';

export default MeetingListItem;
