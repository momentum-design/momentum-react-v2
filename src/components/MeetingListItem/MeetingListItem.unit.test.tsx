import React from 'react';
import { mount } from 'enzyme';

import MeetingListItem, { MEETING_LIST_ITEM_CONSTANTS as CONSTANTS, MeetingMarker } from './';
import ButtonGroup from '../ButtonGroup';
import Icon from '../Icon';
import Avatar from '../Avatar';
import ButtonPill from '../ButtonPill';
import ButtonCircle from '../ButtonCircle';
import { mountAndWait } from '../../../test/utils';
import { render } from '@testing-library/react';
import Text from '../Text';
import ButtonHyperlink from '../ButtonHyperlink';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import List from '../List';

describe('<MeetingListItem />', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(<MeetingListItem />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = mount(<MeetingListItem className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = mount(<MeetingListItem id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = mount(<MeetingListItem style={style} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with isDisabled', () => {
      expect.assertions(1);

      const isDisabled = true;

      const container = mount(<MeetingListItem isDisabled={isDisabled} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with buttonGroup', () => {
      expect.assertions(1);

      const buttonGroup = <ButtonGroup />;

      const container = mount(<MeetingListItem buttonGroup={buttonGroup} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with color', () => {
      expect.assertions(1);

      const color = MeetingMarker.AcceptedActive;

      const container = mount(<MeetingListItem color={color} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with image', async () => {
      expect.assertions(1);

      const image = <Icon name="placeholder" />;

      const container = await mountAndWait(<MeetingListItem image={image} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', () => {
      expect.assertions(1);

      const element = mount(<MeetingListItem />)
        .find(MeetingListItem)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = mount(<MeetingListItem className={className} />)
        .find(MeetingListItem)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      const element = mount(<MeetingListItem id={id} />)
        .find(MeetingListItem)
        .getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = mount(<MeetingListItem style={style} />)
        .find(MeetingListItem)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should have provided data-disabled when isDisabled is provided', () => {
      expect.assertions(1);

      const isDisabled = true;

      const container = mount(<MeetingListItem isDisabled={isDisabled}>Test</MeetingListItem>);

      const element = container.find(MeetingListItem).getDOMNode();

      expect(element.getAttribute('data-disabled')).toBe('true');
    });

    it('should have disable buttons inside if isDisabled is provided', () => {
      expect.assertions(3);

      const isDisabled = true;

      const container = mount(
        <MeetingListItem
          isDisabled={isDisabled}
          buttonGroup={
            <ButtonGroup>
              <ButtonCircle key="1">Test</ButtonCircle>
              <ButtonPill key="2">Test</ButtonPill>
            </ButtonGroup>
          }
        >
          Test
        </MeetingListItem>
      );

      const element = container.find(MeetingListItem).getDOMNode();
      const buttonCircle = container.find(ButtonCircle).getDOMNode();
      const buttonPill = container.find(ButtonPill).getDOMNode();

      expect(element.getAttribute('data-disabled')).toBe('true');
      expect(buttonCircle.getAttribute('data-disabled')).toBe('true');
      expect(buttonPill.getAttribute('data-disabled')).toBe('true');
    });

    it('should have provided data-size', () => {
      expect.assertions(1);

      const container = mount(<MeetingListItem>Test</MeetingListItem>);

      const element = container.find(MeetingListItem).getDOMNode();

      expect(element.getAttribute('data-size')).toBe('50');
    });

    it('should have provided role', () => {
      expect.assertions(1);

      const container = mount(<MeetingListItem>Test</MeetingListItem>);

      const element = container.find(MeetingListItem).getDOMNode();

      expect(element.getAttribute('role')).toBe('listitem');
    });

    it('should have appropriate sizes for button group', async () => {
      expect.assertions(5);

      const container = await mountAndWait(
        <MeetingListItem
          buttonGroup={
            <ButtonGroup>
              <ButtonPill key="1" className="button-pill" />
              <ButtonCircle key="2" className="button-circle">
                <Icon name="chat" />
              </ButtonCircle>
              <Icon key="3" name="placeholder" />
              <Avatar key="4" className="avatar" />
            </ButtonGroup>
          }
        />
      );

      const element = container.find(MeetingListItem).getDOMNode();

      expect(element.getElementsByClassName('button-pill')[0].getAttribute('data-size')).toBe('28');
      expect(element.getElementsByClassName('button-circle')[0].getAttribute('data-size')).toBe(
        '32'
      );
      expect(element.getElementsByTagName('svg')[0].getAttribute('data-scale')).toBe('12');
      expect(element.getElementsByTagName('svg')[1].getAttribute('data-scale')).toBe('12');
      expect(element.getElementsByClassName('avatar')[0].getAttribute('data-size')).toBe('32');
    });
  });

  describe('keyboard navigation', () => {
    it('should work as expected', async () => {
      const user = userEvent.setup();

      const { getByTestId } = render(
        <List listSize={1}>
          <MeetingListItem
            itemIndex={0}
            data-testid="meeting-list-item"
            buttonGroup={
              <ButtonGroup spaced>
                <ButtonHyperlink data-testid="buttonlink" key="link">
                  Link
                </ButtonHyperlink>
                <div key="participants-list" style={{ paddingRight: 0 }}>
                  17
                </div>
                <Icon key="participants-icon" name="participant-list" scale={16} />
                <ButtonPill key="join-button" color="join">
                  Join
                </ButtonPill>
              </ButtonGroup>
            }
            color={MeetingMarker.AcceptedActive}
            image={<Avatar initials="TU" />}
          >
            <Text type="body-primary" key="child1">
              Date
            </Text>
            <Text type="body-secondary" key="child2">
              Normal
            </Text>
          </MeetingListItem>
        </List>
      );

      expect(document.activeElement).toBe(document.body);

      await user.tab();

      const listItem = getByTestId('meeting-list-item');

      expect(listItem).toHaveFocus();

      await user.tab();

      const buttonLink = getByTestId('buttonlink');

      expect(buttonLink).toHaveFocus();
    });
  });
});
