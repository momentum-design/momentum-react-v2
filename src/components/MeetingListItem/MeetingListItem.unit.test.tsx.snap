// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeetingListItem /> snapshot should match snapshot 1`] = `
<MeetingListItem>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    data-color="Transparent"
    isPadded={true}
    size={50}
  >
    <FocusRing
      isInset={false}
    >
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
        isInset={false}
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-allow-text-select={false}
          data-color="Transparent"
          data-disabled={false}
          data-interactive={true}
          data-padded={true}
          data-shape="rectangle"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onPointerDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
          tabIndex={0}
        >
          <MeetingRowContent
            color="Transparent"
          >
            <ListItemBaseSection
              className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
              position="start"
            >
              <div
                className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
                data-position="start"
              >
                <div
                  className="md-meeting-row-content-border"
                  data-color="Transparent"
                />
              </div>
            </ListItemBaseSection>
            <ListItemBaseSection
              className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
              position="middle"
            >
              <div
                className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
                data-position="middle"
              />
            </ListItemBaseSection>
          </MeetingRowContent>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with buttonGroup 1`] = `
<MeetingListItem
  buttonGroup={<ButtonGroup />}
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    data-color="Transparent"
    isPadded={true}
    size={50}
  >
    <FocusRing
      isInset={false}
    >
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
        isInset={false}
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-allow-text-select={false}
          data-color="Transparent"
          data-disabled={false}
          data-interactive={true}
          data-padded={true}
          data-shape="rectangle"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onPointerDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
          tabIndex={0}
        >
          <MeetingRowContent
            buttonGroup={<ButtonGroup />}
            color="Transparent"
          >
            <ListItemBaseSection
              className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
              position="start"
            >
              <div
                className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
                data-position="start"
              >
                <div
                  className="md-meeting-row-content-border"
                  data-color="Transparent"
                />
              </div>
            </ListItemBaseSection>
            <ListItemBaseSection
              className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
              position="middle"
            >
              <div
                className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
                data-position="middle"
              />
            </ListItemBaseSection>
            <ListItemBaseSection
              className="md-meeting-row-content-end-section"
              position="end"
            >
              <div
                className="md-meeting-row-content-end-section"
                data-position="end"
              >
                <ButtonGroup>
                  <div
                    className="md-button-group-wrapper"
                    data-compressed={false}
                    data-orientation="horizontal"
                    data-round={false}
                    data-separator={false}
                    data-spaced={false}
                  />
                </ButtonGroup>
              </div>
            </ListItemBaseSection>
          </MeetingRowContent>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with className 1`] = `
<MeetingListItem
  className="example-class"
>
  <ListItemBase
    className="example-class md-meeting-list-item-wrapper"
    data-color="Transparent"
    isPadded={true}
    size={50}
  >
    <FocusRing
      isInset={false}
    >
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
        isInset={false}
      >
        <li
          className="example-class md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-allow-text-select={false}
          data-color="Transparent"
          data-disabled={false}
          data-interactive={true}
          data-padded={true}
          data-shape="rectangle"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onPointerDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
          tabIndex={0}
        >
          <MeetingRowContent
            color="Transparent"
          >
            <ListItemBaseSection
              className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
              position="start"
            >
              <div
                className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
                data-position="start"
              >
                <div
                  className="md-meeting-row-content-border"
                  data-color="Transparent"
                />
              </div>
            </ListItemBaseSection>
            <ListItemBaseSection
              className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
              position="middle"
            >
              <div
                className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
                data-position="middle"
              />
            </ListItemBaseSection>
          </MeetingRowContent>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with color 1`] = `
<MeetingListItem
  color="AcceptedActive"
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    data-color="AcceptedActive"
    isPadded={true}
    size={50}
  >
    <FocusRing
      isInset={false}
    >
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
        isInset={false}
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-allow-text-select={false}
          data-color="AcceptedActive"
          data-disabled={false}
          data-interactive={true}
          data-padded={true}
          data-shape="rectangle"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onPointerDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
          tabIndex={0}
        >
          <MeetingRowContent
            color="AcceptedActive"
          >
            <ListItemBaseSection
              className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
              position="start"
            >
              <div
                className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
                data-position="start"
              >
                <div
                  className="md-meeting-row-content-border"
                  data-color="AcceptedActive"
                />
              </div>
            </ListItemBaseSection>
            <ListItemBaseSection
              className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-AcceptedActive"
              position="middle"
            >
              <div
                className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-AcceptedActive"
                data-position="middle"
              />
            </ListItemBaseSection>
          </MeetingRowContent>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with id 1`] = `
<MeetingListItem
  id="example-id"
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    data-color="Transparent"
    id="example-id"
    isPadded={true}
    size={50}
  >
    <FocusRing
      isInset={false}
    >
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
        isInset={false}
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-allow-text-select={false}
          data-color="Transparent"
          data-disabled={false}
          data-interactive={true}
          data-padded={true}
          data-shape="rectangle"
          data-size={50}
          id="example-id"
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onPointerDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
          tabIndex={0}
        >
          <MeetingRowContent
            color="Transparent"
          >
            <ListItemBaseSection
              className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
              position="start"
            >
              <div
                className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
                data-position="start"
              >
                <div
                  className="md-meeting-row-content-border"
                  data-color="Transparent"
                />
              </div>
            </ListItemBaseSection>
            <ListItemBaseSection
              className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
              position="middle"
            >
              <div
                className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
                data-position="middle"
              />
            </ListItemBaseSection>
          </MeetingRowContent>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with image 1`] = `
<MeetingListItem
  image={
    <Icon
      name="placeholder"
    />
  }
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    data-color="Transparent"
    isPadded={true}
    size={50}
  >
    <FocusRing
      isInset={false}
    >
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
        isInset={false}
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-allow-text-select={false}
          data-color="Transparent"
          data-disabled={false}
          data-interactive={true}
          data-padded={true}
          data-shape="rectangle"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onPointerDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
          tabIndex={0}
        >
          <MeetingRowContent
            color="Transparent"
            image={
              <Icon
                name="placeholder"
              />
            }
          >
            <ListItemBaseSection
              className="md-meeting-row-content-start-section"
              position="start"
            >
              <div
                className="md-meeting-row-content-start-section"
                data-position="start"
              >
                <div
                  className="md-meeting-row-content-border"
                  data-color="Transparent"
                />
                <Icon
                  name="placeholder"
                  scale={12}
                  strokeColor="none"
                  weight="bold"
                >
                  <div
                    aria-hidden="true"
                    className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-autoscale={false}
                      data-scale={12}
                      data-test="placeholder"
                      fill="currentColor"
                      height="100%"
                      style={
                        {
                          "stroke": "none",
                        }
                      }
                      viewBox="0, 0, 32, 32"
                      width="100%"
                    />
                  </div>
                </Icon>
              </div>
            </ListItemBaseSection>
            <ListItemBaseSection
              className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
              position="middle"
            >
              <div
                className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
                data-position="middle"
              />
            </ListItemBaseSection>
          </MeetingRowContent>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with isDisabled 1`] = `
<MeetingListItem
  isDisabled={true}
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    data-color="Transparent"
    isDisabled={true}
    isPadded={true}
    size={50}
  >
    <FocusRing
      isInset={false}
    >
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
        isInset={false}
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-allow-text-select={false}
          data-color="Transparent"
          data-disabled={true}
          data-interactive={true}
          data-padded={true}
          data-shape="rectangle"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onPointerDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
          tabIndex={0}
        >
          <MeetingRowContent
            color="Transparent"
            isDisabled={true}
          >
            <ListItemBaseSection
              className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
              position="start"
            >
              <div
                className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
                data-position="start"
              >
                <div
                  className="md-meeting-row-content-border"
                  data-color="Transparent"
                />
              </div>
            </ListItemBaseSection>
            <ListItemBaseSection
              className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
              position="middle"
            >
              <div
                className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
                data-position="middle"
              />
            </ListItemBaseSection>
          </MeetingRowContent>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;

exports[`<MeetingListItem /> snapshot should match snapshot with style 1`] = `
<MeetingListItem
  style={
    {
      "color": "pink",
    }
  }
>
  <ListItemBase
    className="md-meeting-list-item-wrapper"
    data-color="Transparent"
    isPadded={true}
    size={50}
    style={
      {
        "color": "pink",
      }
    }
  >
    <FocusRing
      isInset={false}
    >
      <FocusRing
        focusClass="md-focus-ring-wrapper children"
        focusRingClass="md-focus-ring-wrapper children"
        isInset={false}
      >
        <li
          className="md-meeting-list-item-wrapper md-list-item-base-wrapper"
          data-allow-text-select={false}
          data-color="Transparent"
          data-disabled={false}
          data-interactive={true}
          data-padded={true}
          data-shape="rectangle"
          data-size={50}
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onPointerDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="listitem"
          style={
            {
              "color": "pink",
            }
          }
          tabIndex={0}
        >
          <MeetingRowContent
            color="Transparent"
          >
            <ListItemBaseSection
              className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
              position="start"
            >
              <div
                className="md-meeting-row-content-start-section md-meeting-row-content-start-section-no-image"
                data-position="start"
              >
                <div
                  className="md-meeting-row-content-border"
                  data-color="Transparent"
                />
              </div>
            </ListItemBaseSection>
            <ListItemBaseSection
              className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
              position="middle"
            >
              <div
                className="md-meeting-row-content-middle-section md-meeting-row-content-middle-section-Transparent"
                data-position="middle"
              />
            </ListItemBaseSection>
          </MeetingRowContent>
        </li>
      </FocusRing>
    </FocusRing>
  </ListItemBase>
</MeetingListItem>
`;
